src/game/client/water/w3dwatertracks.cpp

```diff
@@ -325,9 +329,213 @@ int WaterTracksObj::Update(int ms_elapsed)
+                }
+            } else {
+                add_point_reset = true;
+            }
+
+            if ((GetAsyncKeyState(VK_DELETE) & 0x8001) != 0) {
+                if (delete_track_reset == true && track != nullptr) {
+                    delete_track_reset = false;
+                    g_theWaterTracksRenderSystem->Un_Bind_Track(track);
+
+                    if (track2 != nullptr) {
+                        g_theWaterTracksRenderSystem->Un_Bind_Track(track2);
+                    }
+
+                    have_start = false;
+                    have_end = false;
+                    track = nullptr;
+                    track2 = nullptr;
+                }
+            } else {
+                delete_track_reset = true;
+            }
+
+            if ((GetAsyncKeyState(VK_INSERT) & 0x8001) != 0) {
+                if (change_type_reset == true) {
+                    change_type_reset = false;
+                    current_wave_type++;
+
+                    if (current_wave_type > WAVE_TYPE_RADIAL) {
+                        current_wave_type = WAVE_TYPE_POND;
+                    }
+
+                    Utf16String str;
+                    str.Format(L"Wave Type: %hs", s_waveTypeInfo[current_wave_type].wave_name);
+                    g_theInGameUI->Message(str);
+                }
+            } else {
+                change_type_reset = true;
+            }
+
+            if ((GetAsyncKeyState(VK_F7) & 0x8001) != 0) {
+                if (save_tracks_reset == true) {
+                    g_theWaterTracksRenderSystem->Save_Tracks();
+                    have_start = false;
+                    have_end = false;
+                    track = nullptr;
+                    track2 = nullptr;
+                    Utf16String str;
+                    str.Format(L"Saved Tracks");
+                    g_theInGameUI->Message(str);
+                }
+            } else {
+                save_tracks_reset = true;
+            }
+
+            if ((GetAsyncKeyState(VK_F8) & 0x8001) != 0) {
+                if (load_tracks_reset == true) {
+                    load_tracks_reset = false;
+                    g_theWaterTracksRenderSystem->Reset();
+                    g_theWaterTracksRenderSystem->Load_Tracks();
+                    have_start = false;
+                    have_end = false;
+                    track = nullptr;
+                    track2 = nullptr;
+                    Utf16String str;
+                    str.Format(L"Loaded Tracks");
+                    g_theInGameUI->Message(str);
+                }
+            } else {
+                save_tracks_reset = true;
+            }
+        }
+
+        if (have_start == true && have_end == true) {
+            ICoord2D coord;
+            coord.x = point.x;
+            coord.y = point.y;
+            g_theTacticalView->Screen_To_Terrain(&coord, &terrain_point_end);
+            int x = terrain_point_end.x - terrain_point_start.x;

```

Comment:
I would consider making this a Coord2D then you could have called .Length() on it instead of sqrt square square